packages/component/src/hooks/internal/useAcknowledgedActivity.js

```diff
@@ -0,0 +1,61 @@
+import { hooks } from 'botframework-webchat-api';
+import { useMemo, useRef } from 'react';
+import { useSticky } from 'react-scroll-to-bottom';
+
+import findLastIndex from '../../Utils/findLastIndex';
+import getActivityUniqueId from '../../Utils/getActivityUniqueId';
+import useChanged from './useChanged';
+
+const { useActivities } = hooks;
+
+// Acknowledged means either:
+// 1. The user sent a message
+//    - We don't need a condition here. When Web Chat send the message, it will scroll to bottom, and it will trigger condition 2 below.
+// 2. At the bottom of the transcript, scrolled from a non-bottom scroll position
+//    - If the transcript is already at the bottom, the user need to scroll up and then go back down
+//    - What happen if we are relaxing "scrolled from a non-bottom scroll position":
+//      1. The condition will become solely "at the bottom of the transcript"
+//      2. Auto-scroll will always scroll the transcript to the bottom
+//      3. Web Chat will always acknowledge all activities as it is at the bottom
+//      4. Acknowledge flag become useless
```

Comment:
add "therefore... + explanation"

Reply:
Added:

```js
//      5. Therefore, even the developer set "pause after 3 activities", if activities are coming in at a slow pace (not batched in a single render)
//         Web Chat will keep scrolling and not snapped/paused
```

Reply:
Don't see 5. added to the code yet, but looks good.
```suggestion
//      5. Therefore, when the developer sets "pause after 3 activities", if activities are coming in at a slow pace (not batched in a single render)
//         Web Chat will keep scrolling and not become snapped/paused
```